### PR TITLE
Update

### PR DESCRIPTION
**Pull request by oleksiivorobiov**

### CODE REVIEW

File: src/mongoc/cnv-mongoc-client-pool.c [hunk 1]
```diff
@@ -160,64 +135,80 @@ cnv_mongoc_client_pool_pop (cnv_mongoc_client_pool_t *pool)
...
      }
 #endif
-   } else {     
-     mongoc_mutex_unlock(&pool->mutex);
+   } else {
      client = pooledClient->client;
      bson_free(pooledClient);
    }
 
+   mongoc_mutex_unlock(&pool->mutex);
```
Comment (jsbattig) [inline src/mongoc/cnv-mongoc-client-pool.c:154]: Alex, you move the lock out here?
I don't want building connection if connect happens at that time inside lock

Comment (oleksiivorobiov) [inline src/mongoc/cnv-mongoc-client-pool.c:154]: Notice there is additional argument here pool->topology, this is shared data and probably it requires locking. I'll check implementation to see how it's used

Comment (jsbattig) [inline src/mongoc/cnv-mongoc-client-pool.c:154]: I see.
You should check what are the operations performed in topology. By the name I'm guessing it probably tries to assert what is the replica/sharding topology dictated by the URI is. If the topology object stays invariant after creation, we can use it outside of the lock.

Comment (jsbattig) [inline src/mongoc/cnv-mongoc-client-pool.c:154]: @alexeyvo , I will merge for now so Fede can keep moving forward, but let's take a look if you can create the connection outside of the lock.

**Pull request merged**